src/routes/safe_apps/handlers.rs

```diff
@@ -5,9 +5,16 @@ use crate::routes::safe_apps::models::SafeApp;
 use crate::utils::context::RequestContext;
 use crate::utils::errors::ApiResult;
 
-pub async fn safe_apps(context: &RequestContext, chain_id: &String) -> ApiResult<Vec<SafeApp>> {
-    let url = config_uri!("/v1/safe-apps/?chainId={}", chain_id);
-
+pub async fn safe_apps(
+    context: &RequestContext,
+    chain_id: &String,
+    client_url: &Option<String>,
+) -> ApiResult<Vec<SafeApp>> {
+    let url = config_uri!(
+        "/v1/safe-apps/?chainId={}&clientUrl={}",
+        chain_id,
+        client_url.clone().unwrap_or("".to_string())
```

Comment:
if you get rid of `to_string` do you still need to `clone`?

Reply:
If I remove `to_string` then I get this error: https://stackoverflow.com/questions/32500974/expected-string-found-str-when-matching-an-optional-string

Reply:
I think here maybe doing `client_url.as_deref().unwrap_or("")` is a bit better, as this will both remove your `&` from the `Option` and return you `&str` after the unwarp.

Reply:
Thanks, I've just updated. I'm curious what's the difference?

Reply:
because your input is `&Option<String>` you were forced to call unwrap with `"".to_string()`. [as_deref](https://doc.rust-lang.org/std/option/enum.Option.html#method.as_deref) will change our type from that to `Option<&String>` and since, `&String` and `&str` are more or less interchangably used, then your `unwrap("")` no longer needs to call `to_string` as `""` is of type `&str`

You can read more about it here:
- https://users.rust-lang.org/t/whats-the-difference-between-string-and-str/10177/2